src/Dijkstra/AST/Examples/PaperIntro.agda

```diff
@@ -0,0 +1,165 @@
+{- Byzantine Fault Tolerant Consensus Verification in Agda, version 0.9.
+
+   Copyright (c) 2022 Oracle and/or its affiliates.
+   Licensed under the Universal Permissive License v 1.0 as shown at https://opensource.oracle.com/licenses/upl
+-}
+module Dijkstra.AST.Examples.PaperIntro (Ev Wr St : Set) where
+
+open import Data.List                   using (List ; length ; [] ; _∷_ ; _++_)
+open import Data.Maybe                  using (Maybe ; maybe ; just ; nothing)
+open import Data.Product                using (_×_ ; _,_ ; proj₁ ; proj₂)
+open import Data.Unit.NonEta            using (Unit; unit)
+open import Dijkstra.AST.RWS Ev Wr St
+-- TODO-2: We should not be depending on things in the Haskell namespace here as Dijkstra.*
+-- should be independent of things outside the Dijkstra namespace
+open import Haskell.Prelude             using (_>>_ ; _>>=_ ; return)
+open import Relation.Binary.PropositionalEquality
+
+-- The example in the paper
+prog : (St → Maybe Wr) → RWSAST Unit
+prog f = pass inner
+  where inner : RWSAST (Unit × (List Wr → List Wr))
+        inner = do
+          m ← gets f
+          maybeAST (λ w → do tell (w ∷ [])
+                             return (unit , λ _ → []))
+                   (return (unit , (λ x → x ++ x)))
+                   m
+
+ProgPost : (Ev × St) → (Unit × St × List Wr) → Set
+ProgPost (_ , s1) (_ , s2 , o) = s1 ≡ s2 × 0 ≡ length o
+
+-- Here we prove that, for any f and i, the postcondition holds of the result of running (prog f)
```

Comment:
change to:

"Here we prove (_without_ using the framework) that, ..."

Reply:
DONE in #239